GLOSSARY.md

```diff
@@ -0,0 +1,167 @@
+formula. For example the goal `exists<T> { T: u32 }` can be solved with `T =
+u32`.
+
+## Literal
+A literal is an atomic element of a formula together with the constants `true`
+and `false`. It is equivalent to a variable in an algebraic expressions. Note
+that literals are *not* the same as the type variables used in specifying a
+goal.
+
+## Normal form
+To say that a statement is in a certain *normal form* means that the pattern in
+which the subformulas are arranged fulfil certain rules. The individual patterns
+have different advantages for their manipulation.
+
+### Conjunctive normal form (CNF)
+A formula in CNF is a disjunction of conjunctions. For example `(x1 || x2 ||
+x3) && (x4 || x5 || x6)` is in CNF.
+
+### Disjunctive normal form (DNF)
+A formula in DNF is a conjunction of disjunctions. For example `(x1 && x2 &&
+x3) || (x4 && x5 && x6)` is in DNF.
+
+### Negation normal form (NNF)
+A formula in NNF consists only of literals, the connectives `&&` and `||` and
+`true` and `false`.
+
+### Prenex normal form (PNF)
+All quantifiers are on the highest level of a formula and do not occur inside
+the subformulas of the expression.
+
+- `forall(x). exists(y). forall(z). P(x) && P(y) => P(z)` is in PNF.
+- `(exists(x). P(x)) => exists(y). P(y) && forall(z). P(z)` is *not* in PNF.
+
+## Normalization
+Normalization is the process of converting an associated type to a concrete
+type. In the case of an iterator this would mean that the associated `Item` type
+is replaced with something more meaningful with respect to the individual
+context (e.g. `u32`).
+
+## Projection
+Projection is the reference to a field or (in the context of Rust) to a type.
+
+## Unification
+Unification is the process of solving a formula. That means unification finds
+values for all the free literals of the formula that satisfy it. In the context
+of chalk the values refer to types.
+
+## Universe
+(empty)
+
+## Well-formed
+(empty)
+
+## Quantifier
+
+### Existential quantifier
+A formula with the existential quantifier `exists(x). P(x)` is satisfiable if
+and only if there exists at least one value for all possible values of x which
+satisfies the subformula `P(x)`.
+
+In the context of chalk, the existential quantifier usually demands the
+existence of exactly one instance (i.e. type) that satisfies the formula (i.e.
+type constraints). More than one instance means that the result is ambiguous.
+
+
+### Universal quantifier
+A formula with the universal quantifier `forall(x). P(x)` is satisfiable
+if and only if the subformula `P(x)` is true for all possible values for x.
+
+### Helpful equivalences
+- `not(forall(x). P(x)) <=> exists(x). not(P(x))`
+- `not(exists(x). P(x)) <=> forall(x). not(P(x))`
+
+## Skolemization
+Skolemization is a technique of transferring a logical formula with existential
+quantifiers to a statement without them. The resulting statement is in general
+not equivalent to the original statement but equisatisfiable.
```

Comment:
Thank you for your thorough explanations! I will work them in during the next days :)